### PR TITLE
Fix Sentry initialization on Amazon Linux

### LINKED ISSUE #635: Sentry does not work on dedicated server based on Amazon_Linux_2023 and Amazon_Linux_2
### Environment
Amazon_Linux_2023 and Amazon_Linux_2

### How do you use Sentry?
Just start dedicated server on AWS instance.

### Which version of the SDK?
SENTRY_SDK_VERSION "0.7.6"

### How did you install the package? (Git-URL, Assetstore)
Git-URL(Not-marketplace)

### Which version of Unreal?
5.3

### Is this happening in Unreal (editor) or on a player like Android, iOS, Windows?
All good. Works perfect on Windows.

### Steps to Reproduce

1. Build Dedicated server and upload to AWS
2. Start server

### Expected Result

Sentry works and initializes fine. 

### Actual Result
Sentry fails to initialize
Image

What I found out that - 
```
ldd crashpad_handle give us:
libc++.so.1 => not found
libc++abi.so.1 => not found
```

### PR DESCRIPTION
**Pull request by tustanivsky**
Currently `sentry-native` is built using `clang` and `libc++` to make the corresponding static libs included in `sentry-unreal` plugin compatible with UE. This introduces additional overhead for using Sentry on Linux since `libc++` as one of the `Crashpad` dependencies has to be installed separately and for some distros this package is not available (i.e. AL 2/2023).

This PR suggests to build `sentry-native` in CI twice: once to obtain static libs using the current configuration and once to get the `Crashpad` executable not dependent on `libc++` using `gcc` which then replaces binary obtained during the first build.

Related #431 

Closes #635

### CODE REVIEW

**Comment by tustanivsky** [comment]
> I think I'm still stuck on the problem conceptually: We can build `sentry-native` statically using `clang & libc++`. But the `crashpad handler` that gets build alongside it has the `libc++` dependency?

@bitsandfoxes Yes, the `crashpad_handler` as it's currently built for Unreal has a dependency on `libc++`. To my understanding the existing `sentry-native` build scripts do not allow building crashpad executable separately with a configuration (compiler, standard library) that's different from the remaining SDK. So we came up with this "hack" when building things twice and combining the resulting artifacts allow us to avoid potential issues with additional deployment environment setup.

cc @supervacuus

File: scripts/build-linux.sh [hunk 1]
```diff
@@ -6,11 +6,21 @@ export sentryArtifactsDestination=$2
...
 cmake -S "${sentryNativeRoot}" -B "${sentryNativeRoot}/build" -D SENTRY_BACKEND=crashpad -D SENTRY_TRANSPORT=none -D SENTRY_SDK_NAME=sentry.native.unreal -D SENTRY_BUILD_SHARED_LIBS=OFF \
     -D CMAKE_BUILD_TYPE=RelWithDebInfo -D CMAKE_C_COMPILER=clang -D CMAKE_CXX_COMPILER="clang++" -D CMAKE_CXX_FLAGS="-stdlib=libc++" -D CMAKE_EXE_LINKER_FLAGS="-stdlib=libc++"
 cmake --build "${sentryNativeRoot}/build" --target sentry --parallel
 cmake --install "${sentryNativeRoot}/build" --prefix "${sentryNativeRoot}/install"
 
+# Build sentry-native using gcc for crashpad which's not depending on libc++
+cmake -S "${sentryNativeRoot}" -B "${sentryNativeRoot}/build_crashpad_handler" -D SENTRY_BACKEND=crashpad -D SENTRY_TRANSPORT=none -D SENTRY_SDK_NAME=sentry.native.unreal -D SENTRY_BUILD_SHARED_LIBS=OFF \
+    -D CMAKE_BUILD_TYPE=RelWithDebInfo
+cmake --build "${sentryNativeRoot}/build_crashpad_handler" --target sentry --parallel
+cmake --install "${sentryNativeRoot}/build_crashpad_handler" --prefix "${sentryNativeRoot}/install_crashpad_handler"
```
[the remark this answers is not in the document]
Comment (supervacuus) [reply scripts/build-linux.sh:19]: > I take it there is no way to simply build the crashpad handler itself outside of sentry-native?

There is. No one forces you to use the top-level build script in the Native SDK. You can only build the `crashpad_handler` target from the crashpad build script we provide (i.e., entirely ignoring the Native SDK), and you can also build only the `crashpad_handler` target from the top-level build script, as shown here: https://discord.com/channels/621778831602221064/1293686979166474372/1293958457174786130

There is also no reason to build with `gcc`. `clang` can also target `libstdc++` as shown in the snippet above.

> But I feel like we should put a big, fat hack in the comments somewhere.

Given that this Unreal Linux build _requires_ two different toolchain configurations, I would like to hear a rational argument for this statement. It requires `clang` but deploys to a target that does not host its standard library (the Native SDK build above also doesn't link statically against a standard library).

**Comment by supervacuus** [comment]
> I think I'm still stuck on the problem conceptually: We can build sentry-native statically using clang & libc++. But the crashpad handler that gets build alongside it has the libc++ dependency?

Everything you're building in this script has the `libc++` dependency since you switched to `clang`, including `libsentry.[a|so]`. 

Building _any library_ statically does not mean the `libc++` dependency is gone; it is not linked statically to a standard library (and it is also unclear how much sense it would make in such a situation). 

The difference is that this dependency is resolved for `libsentry` as part of the Unreal executable, whereas the `crashpad_handler` is a separate executable and gets the standard library the OS loader looks up for it.

**Review by bitsandfoxes — APPROVED** [review]
I would really like to see this covered by some smoke test. I.e. run that amazon docker image in CI?

**Comment by tustanivsky** [comment]
We should be able to add smoke tests once #658 is resolved.

**Pull request merged**